jax/_src/util.py

```diff
@@ -34,16 +34,18 @@
 T = TypeVar("T")
 
 def safe_zip(*args):
-  n = len(args[0])
-  for arg in args[1:]:
-    assert len(arg) == n, 'length mismatch: {}'.format(list(map(len, args)))
+  if len(args) != 0:
```

Comment:
A benefit of the previous behavior was that `safe_zip` errored with empty arguments, which is sometimes an indication of something unexpected happening. What if we handle the empty padding case without modifying `safe_zip`?

Reply:
I personally think that the new behavior would be a bit better in that it would be consistent with the behavior of `map` and `zip`, as well as logically correct (an empty set satisfies any condition), so I think it would be better if empty arguments, if they are unexpected, would cause errors elsewhere, but not inside these methods. But I'm not that familiar with JAX codebase, so if you still prefer these to fail on empty entires, happy to revert the changes to both `safe_map` and `safe_zip`, let me know!

Reply:
Looping in @mattjj because I think he has opinions...